### DISCUSSION TITLE
/gems/ruby-vips-2.1.4/lib/vips/operation.rb:218:in `initialize': VipsOperation: class "gifsave" not found (Vips::Error)

### DISCUSSION THREAD

**Discussion by serhiihiba** [libvips/ruby-vips#354] [General]
On macos it works, on ubuntu not

**Comment by jcupitt** [comment]
Hi @enotikalt, 

`gifsave` relies on the `cgif` library, so you need to build libvips against that:

https://github.com/dloebl/cgif

brew does this, but the standard ubuntu package does not. You'll need to build libvips yourself if you need this feature.

You can use `write_to_file` instead: this will examine the filename suffix and pick the best saver it can find. Without cgif, this will fall back to the slower and more memory hungry imagemagick.

**Reply by serhiihiba** [reply]
I'm not sure I understand, how I can build `libvips` with `cgif`?

**Reply by jcupitt** [reply]
You'd need to compile the thing from source.
